### PR TITLE
feat: Admin V2 API keys

### PR DESCRIPTION
**Pull request by olivermrbl**
Add API key management UI for V2

This PR only adds publishable API key UI. Secret API key management will come in a follow-up PR.

### CODE REVIEW

File: packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-add-sales-channels/api-key-management-add-sales-channels.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    [adminPublishableApiKeysKeys.detailSalesChannels(id!)],
+    {
+      fields: "id,sales_channels",
+    },
+    {
+      keepPreviousData: true,
+    }
+  )
+
+  const salesChannels = data?.api_key?.sales_channels
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-add-sales-channels/api-key-management-add-sales-channels.tsx]: suggestion: maybe should be moved to after the error check

File: packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-detail/components/api-key-sales-channel-section/api-key-sales-channel-section.tsx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+    [pageIndex, pageSize]
+  )
+
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({})
+
+  const params = useQueryParams(["q"])
+
+  const query = {
+    ...params,
+    fields: "id,sales_channels",
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-detail/components/api-key-sales-channel-section/api-key-sales-channel-section.tsx]: Q: shouldn't it be `id,*sales_channels`

File: packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-edit/api-key-management-edit.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  const { data, isLoading, isError, error } = useAdminCustomQuery(
+    `/api-keys/${id}`,
+    [adminPublishableApiKeysKeys.detail(id!)]
+  )
+
+  if (isError || !data?.api_key) {
+    if (error) {
+      throw error
+    }
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-edit/api-key-management-edit.tsx:20]: todo: I believe we are missing the handling of no api key here

File: packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-edit/api-key-management-edit.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  return (
+    <RouteDrawer>
+      <RouteDrawer.Header>
+        <Heading>{t("apiKeyManagement.editKey")}</Heading>
+      </RouteDrawer.Header>
+      {!isLoading && data?.api_key && <EditApiKeyForm apiKey={data?.api_key} />}
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-edit/api-key-management-edit.tsx]:
```suggestion
      {!isLoading && !!data?.api_key && <EditApiKeyForm apiKey={data.api_key} />}
```

File: packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-filters.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  const dateFilters: Filter[] = [
+    { label: t("fields.createdAt"), key: "created_at" },
+    { label: t("fields.updatedAt"), key: "updated_at" },
+    { label: t("fields.revokedAt"), key: "revoked_at" },
+  ].map((f) => ({
+    key: f.key,
+    label: f.label,
+    type: "date",
+  }))
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-filters.tsx]:
```suggestion
  const dateFilters: Filter[] = [
    { label: t("fields.createdAt"), key: "created_at",  type: "date" },
    { label: t("fields.updatedAt"), key: "updated_at",  type: "date" },
    { label: t("fields.revokedAt"), key: "revoked_at",  type: "date" },
  ]
```

Comment (adrien2p) [reply packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-filters.tsx]: I believe it is easier to understand

File: packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-query.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    ["offset", "q", "created_at", "updated_at", "revoked_at", "order"],
+    prefix
+  )
+
+  const { offset, created_at, updated_at, revoked_at, q, order } = queryObject
+
+  const searchParams: GetPublishableApiKeysParams = {
+    limit: pageSize,
+    offset: offset ? Number(offset) : 0,
+    created_at: created_at ? JSON.parse(created_at) : undefined,
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-query.tsx]: Q: why do we need to parse the data here?

**Review by adrien2p — APPROVED** [review]
Very nice 🚀 few suggestions and questions

File: packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-query.tsx (hunk 6 above)
Comment (olivermrbl) [reply packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-query.tsx]: Not sure, we've always done that. Maybe @kasperkristensen has the answer for this 👍

File: packages/admin-next/dashboard/public/locales/en-US/translation.json [hunk 7]
```diff
@@ -817,6 +817,21 @@
     "createdBy": "Created by",
     "revokedBy": "Revoked by"
   },
+  "apiKeyManagementV2": {
```
Comment (sradevski) [inline packages/admin-next/dashboard/public/locales/en-US/translation.json]: I guess now we can remove the v2 prefix

**Review by sradevski — APPROVED** [review]
Looks good!

**Comment by olivermrbl** [comment]
I'll add proper typing once the new hook tool and fetch client have been introduced :)

**Pull request merged**